src/core/speechrecognition/locales.js

```diff
@@ -9,15 +10,39 @@ import { SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE } from '../constants';
  * @param {string} locale
  * @returns {string}
  */
-export function transformSpeechRecognitionLocaleForEdge (locale) {
-  const underscoreIndex = locale.indexOf('_');
-  if (underscoreIndex === -1) {
-    return locale;
+export function transformSpeechRecognitionLocaleForEdge (rawLocale) {
+  const { language, modifier, region } = parseLocale(rawLocale);
+  if (!modifier && !region) {
+    return language;
   }
-  locale = locale.replace('_', '-');
-  const isCompatibleWithEdge = SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE.includes(locale.toLowerCase());
+  const locale = formatLocaleForEdge(language, modifier, region);
+  const isCompatibleWithEdge = SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE.includes(locale);
   if (isCompatibleWithEdge) {
     return locale;
   }
-  return locale.substring(0, underscoreIndex);
+  if (modifier) {
+    return formatLocaleForEdge(language, modifier);
+  }
```

Comment:
If a locale isn't in the compatibility list but it has a modifier, we return it? I would expect that if the locale isn't in the compatibility list, we return just the language, not a language and modifier.

Reply:
If zh-hans-bob isn't supported, we can return zh-hans instead of just zh
by default edge will interpret zh on its own as zh-hans, so this only matters for zh-hant-[unsupport-region]

Reply:
alright, got it